source/features/toggle-everything-with-alt.tsx

```diff
@@ -0,0 +1,83 @@
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import features from '../libs/features';
+
+function getSimilarItems(item: HTMLElement): HTMLElement[] {
+	// Collapsed comments in PR conversations and files
+	if (item.matches('summary')) {
+		if ((item.parentElement as HTMLDetailsElement).open) {
+			return select.all('.minimized-comment details[open] summary');
+		}
+
+		return select.all('.minimized-comment details:not([open]) summary');
+	}
+
+	// "Load diff" buttons in PR files
+	if (item.matches('.js-diff-load')) {
+		return select.all('.js-file .js-diff-load');
+	}
+
+	// Review comments in PR
+	if (item.matches('.js-resolvable-thread-toggler')) {
+		const targets = select.all('.js-resolvable-thread-toggler');
+		if (select.exists('svg.octicon-unfold', item)) {
+			return targets.filter(target => !target.matches('.d-none') && select.exists('svg.octicon-unfold', target));
+		}
+
+		return targets.filter(target => !target.matches('.d-none') && select.exists('svg.octicon-fold', target));
+	}
+
+	// "Show comments" checkboxes
+	if (item instanceof HTMLLabelElement) {
+		const inputs = select.all<HTMLInputElement>('.js-file .dropdown-item .js-toggle-file-notes');
+		if ((item.control as HTMLInputElement).checked) {
+			return inputs.filter(input => !input.checked).map(input => input.labels![0]);
+		}
+
+		return inputs.filter(input => input.checked).map(input => input.labels![0]);
+	}
+
+	return [];
+}
+
+function handleEvent(event: DelegateEvent<MouseEvent, HTMLElement>): void {
+	if (!event.altKey) {
+		return;
+	}
+
+	const clickedItem = event.delegateTarget;
+	const viewportOffset = clickedItem.parentElement!.getBoundingClientRect().top;
+	const similarItems = getSimilarItems(clickedItem);
+
+	for (const item of similarItems) {
+		if (item !== clickedItem) {
+			item.click();
+		}
+	}
```

Comment:
I don't think we should ever cause multi-second locks, it's just bad, especially because it will look like the page is unresponsive and the user might alt-click it again and... now the queue is longer.

---

It's a little more complicated because each batch would have to be surrounded by a `get scroll` and `set scroll`, but the `anchorScroll` we have in #2146 should help. 

```js
const chunked = chunkArray(similarItems.filter(item => item !== clickedItem), 3);
for (const chunk of chunked) {
	await anchorScroll(() => {
		for (const item of chunked) {
			item.click();
		}
	});
}
```

Or maybe an easier way to just process 3 at a time with a regular for loop

Reply:
> It's a little more complicated because each batch would have to be surrounded by a `get scroll` and `set scroll`

Yup, that's the problem. But its more like guarding each call to `.click()` and its respective layout change, while handling scroll position changes.

> but the `anchorScroll` we have in #2146 should help.

That doesn't seem to help much too, the scroll position ends up being at some random positions after triggering the clicks.

I've tried handling duplicate events, any events emitted on repeated clicks, chunking callbacks, even plain promises, and all of them have the same symptom, the scroll position ends up being at random positions (especially for chunked triggers and allowing user scrolling without jank).

Reply:
> the scroll position ends up being at some random positions after triggering the clicks

Indeed, our function only works with sync changes but in this case the fragments might be loaded asynchronously so our "reset position on rAF" was **before** the fragment loaded. I dropped it in 867752c2

Anyway it looks like the browser's scroll anchoring already takes care of it if the elements are completely outside the viewport.